Documentation/network/node-ipam.rst

```diff
@@ -42,3 +42,101 @@ To enable the node IPAM on an existing installation, run:
+to allow all Nodes in the Cluster to listen for incoming connections. Setting the ``loadBalancerClass`` to ``io.cilium/node``
+will add all Nodes to the Ingress IP list in the Load Balancer Service and accept traffic on the requested ports.
+
+Selecting Nodes to Listen for Gateway and Ingress Requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To restrict the Nodes that should listen for incoming traffic for Cilium Gateway and Ingress Services, you may
+add an annotation, ``io.cilium.nodeipam/match-labels``, to the Service. The value of the annotation is a
+`Label Selector <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors>`__.
+
+.. note::
+   The node selector feature only works for Cilium-managed LoadBalancers for Gateways and Ingresses.
+   Applying this label has no effect on other LoadBalancer Services.
+
+
+You can test this Label Selector by using the ``kubectl get nodes`` command. For example, to exclude traffic to all
+Control Plane nodes, you may use this Label Selector: ``!node-role.kubernetes.io/control-plane``
+The Annotation would be configured as follows:
+
+.. parsed-literal::
+   io.cilium.nodeipam/match-labels="!node-role.kubernetes.io/control-plane"
+
+To verify the Nodes that would be returned, run the following command in your terminal:
+
+.. parsed-literal::
+   kubectl get nodes -l '!node-role.kubernetes.io/control-plane'
+
+   NAME          STATUS   ROLES    AGE     VERSION
+   kind-worker   Ready    <none>   6h32m   v1.29.1
+
+It is possible to combine multiple filters. For example, Nodes where the Label ``beehive.local/node`` value is
+``hive1`` or ``hive2``, the Label ``honey`` is set to ``yes``, and the Label ``bears`` does not exist would use the
+following filter Annotation:
+
+.. parsed-literal::
+   io.cilium.nodeipam/match-labels="beehive.local/node in (hive1,hive2),honey=yes,!bears"
+
+Configuring Resources for Node IPAM LB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Ingress Controller**
+
+Use the following Helm values to enabling Node IPAM LB for the shared Ingress Controller:
+
+.. parsed-literal::
+   nodeIPAM:
+     enabled: true
+   ingressController:
+     enabled: true
+     loadbalancerMode: shared
+     service:
+       loadBalancerClass: io.cilium/node
+   # Annotation is optional. Enable to limit Load Balancer IPs to the desired Nodes.
+       annotations:
+         io.cilium.nodeipam/match-labels: "..."
+
+Node IPAM LB support for dedicated Ingress resources is not directly possible.
+See the Warning in the Gateway API section below for more details on how to possibly configure a policy to update
+your dedicated Ingresses. You will need to add the ``io.cilium.nodeipam`` prefix to the ``ingressController.ingressLBAnnotationPrefixes``
+value in your Helm Chart.
+
+Using Node IPAM LB with dedicated Ingresses has not been tested!
```

Comment:
I am not very aware of this mode but I would guess that with your patch it should work as is, right? if it is as simple as changing the `loadbalancerMode` to `dedicated` above to test that theory, could you confirm if it working or not and change this sentence depending of the result of this :pray:?